src/cs_encrypted_v1.schema.json

```diff
@@ -15,7 +15,7 @@
     "k": {
       "title": "kind",
       "type": "string",
-      "enum": ["pt", "ct", "en"]
+      "enum": ["pt", "ct", "en", "qm", "qo", "qu", "qejp", "qsv"]
```

Comment:
When "for query" is set on the plaintext payload, Proxy will encrypt the payload as one of the new variants here:
- `qm` = query match
- `qo` = query ORE
- `qu` = query unique
- `qejp` = query EJSON path
- `qsv`= query STE vec

Proxy will only generate the index needed for the specific query and set the relevant kind/index field in the payload.

For example, if the client sends through a plaintext payload with "for query" set to "match":
```js
{
  "v": 1,
  "k": "pt",
  "p": "abc",
  "i": {
    "t": "users",
    "c": "email_encrypted"
  },
  "q": "match"
}
```

Proxy would encrypt that as the match query variant:
```
{
  "v": 1,
  "k": "qm",
   "m": [1, 2, 3]
}
```

Note that the `"m"` key in the match query payload lines up with the same key used for the match index that's stored in the column. This allows the EQL SQL funcs to look up the index field using the same key (whether the value comes from the column or from a param):
```sql
-- $1 here should be encrypted for a match query
SELECT * FROM users WHERE cs_match_v1(email_encrypted) @> cs_match_v1($1)
```

Reply:
Are the new `kinds` required? 
The postgres function shouldn't need the kind as long as the actual field is encoded correctly.

Reply:
They aren't strictly required (we could re-use the existing `"ct"`/ciphertext variant), but I'd prefer to model this as new variants for a few reasons.

On the Proxy/Rust side:
- The source ciphertext (`"c"` field) in the existing `"ct"`/ciphertext variant would have to become optional (since queries wouldn't use it)
- All of the indexes are optional in the existing `"ct"`/ciphertext variant, which is bit awkward to work with when encrypting for a query because behaviour-wise only one (required) index field should be set in the payload depending on the operation
- Basically, specific variants for each makes it harder for us to construct something incorrectly and means that we can lean more on the compiler to catch problems

On the PG side:
- If PG attempts to store a `q*` variant, something has gone wrong. It's convenient to be able to reject anything that isn't the `ct` variant
- It's nice for debugging to be able to be able to identify with specific variant was chosen

Reply:
I think having variants for every option adds complexity. 
We could have a single "query" variant to disambiguate from ct and pt.

Coupled with having the "q" value be the identifier, we can then validate. 

```
"knd": "qt", 
"qry": "u",  
"u": "etc",
```

if knd == qt
  qry is required and qry.value (u) should exist as a field.

Reply:
I think that would be an OK middle ground if we're concerned about the number of variants.

Conceptually there's a variant for each and I rather make that explicit than have to model it as a single type that requires extra validation logic to handle what the compiler could catch for us.